### DISCUSSION TITLE
Realization of SF (Slicing Aided Fine-tuning)  module and design of related parameters

### DISCUSSION THREAD

**Discussion by YipChakwai** [fcakyon/small-object-detection-benchmark#25] [Q&A] [answered]
Hello sir, I would like to know whether the SF(Slicing Aided Fine-tuning) module is implemented by the train_pipeline in the [mmdet_configs](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/mmdet_configs/visdrone_fcos/fcos_crop_480_960_cls_60.py) file? 
`train_pipeline = [
    dict(type="LoadImageFromFile"),
    dict(type="LoadAnnotations", with_bbox=True),
    dict(
        type="AutoAugment",
        policies=[
            [
                dict(type="RandomCrop", crop_type="absolute_range", crop_size=(480, 960), allow_negative_crop=True),
                dict(type="Resize", img_scale=(1333, 800), keep_ratio=True),
            ],
            [
                dict(type="RandomCrop", crop_type="absolute_range", crop_size=(480, 960), allow_negative_crop=True),
                dict(type="Resize", img_scale=(1333, 800), keep_ratio=True),
            ],
            [
                dict(type="Resize", img_scale=(1333, 800), keep_ratio=True),
            ],
        ],
    ),
    dict(type="RandomFlip", flip_ratio=0.5),
    dict(type="Normalize", **img_norm_cfg),
    dict(type="Pad", size_divisor=32),
    dict(type="DefaultFormatBundle"),
    dict(type="Collect", keys=["img", "gt_bboxes", "gt_labels"]),
]`
 If yes, how should the parameter crop_size be designed? Is it designed according to the size of visdrone and xview dataset pictures?

**Comment by fcakyon** [comment, marked as answer]
> Hello sir, I would like to know whether the SF(Slicing Aided Fine-tuning) module is implemented by the train_pipeline in the [mmdet_configs](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/mmdet_configs/visdrone_fcos/fcos_crop_480_960_cls_60.py) file? 

Yes.

> If yes, how should the parameter crop_size be designed? Is it designed according to the size of visdrone and xview dataset pictures?

Yes.

You have to arrange the slice sizes considering your object to image ratio (if average size of objects/average size of images is very low you can use smaller slices.)

**Reply by YipChakwai** [reply]
Thank you for your answer. Should the size of the sliced pictures in SF and SAHI be the same? If the picture is randomly sliced during training (type="RandomCrop", crop_type="absolute_range"), how to design a fixed slice size when performing sahi predict?

**Reply by fcakyon** [reply]
You can use average size or max size during inference.

**Reply by YipChakwai** [reply]
Ok, I'll try. Thank you for your patience！
